utils_nlp/eval/rouge_ext.py

```diff
@@ -0,0 +1,978 @@
+              Default:665 (bytes)
+          length_limit_type: Unit of length_limit. Available: words, bytes. Default: 'bytes'
+          apply_avg: If we should average the score of multiple samples. Default: True. If
+              apply_Avg & apply_best = False, then each ROUGE scores are independant
+          apply_best: Take the best instead of the average. Default: False, then each ROUGE
+              scores are independant
+          stemming: Apply stemming to summaries. Default: True
+          alpha: Alpha use to compute f1 score: P*R/((1-a)*P + a*R). Default:0.5
+          weight_factor: Weight factor to be used for ROUGE-W. Official rouge score defines
+              it at 1.2. Default: 1.0
+
+        Raises:
+          ValueError: raises exception if metric is not among AVAILABLE_METRICS
+          ValueError: raises exception if length_limit_type is not among
+              AVAILABLE_LENGTH_LIMIT_TYPES
+          ValueError: raises exception if weight_factor < 0
+        """
+        self.metrics = metrics[:] if metrics is not None else RougeExt.DEFAULT_METRICS
+        for m in self.metrics:
+            if m not in RougeExt.AVAILABLE_METRICS:
+                raise ValueError("Unknown metric '{}'".format(m))
+
+        self.max_n = max_n if "rouge-n" in self.metrics else None
+        # Add all rouge-n metrics
+        if self.max_n is not None:
+            index_rouge_n = self.metrics.index("rouge-n")
+            del self.metrics[index_rouge_n]
+            self.metrics += ["rouge-{}".format(n) for n in range(1, self.max_n + 1)]
+        self.metrics = set(self.metrics)
+
+        self.limit_length = limit_length
+        if self.limit_length:
+            if length_limit_type not in RougeExt.AVAILABLE_LENGTH_LIMIT_TYPES:
+                raise ValueError("Unknown length_limit_type '{}'".format(length_limit_type))
+
+        self.length_limit = length_limit
+        if self.length_limit == 0:
+            self.limit_length = False
+        self.length_limit_type = length_limit_type
+        self.stemming = stemming
+
+        self.apply_avg = apply_avg
+        self.apply_best = apply_best
+        self.alpha = alpha
+        self.weight_factor = weight_factor
+        if self.weight_factor <= 0:
+            raise ValueError("ROUGE-W weight factor must greater than 0.")
+
+        self.language = language
+
+        self.sentence_split = RougeExt.SENTENCE_SPLIT_DICT[self.language]
+        self.word_tokenize = RougeExt.WORD_TOKENIZE_DICT[self.language]
+        self.remove_char_pattern = RougeExt.REMOVE_CHAR_PATTERN_DICT[self.language]
+        if self.language not in RougeExt.STEMMER_DICT.keys():
+            self.stemmer = None
+            warnings.warn("Language-specific stemmer is not available. Skipping stemming.")
+        else:
+            self.stemmer = RougeExt.STEMMER_DICT[self.language]
+
+        # # Load static objects
```

Comment:
can we just remove these commented-out code lines?

Reply:
I prefer keeping these to keep record of what I changed to the original code. I found this useful when I looked at how BERTSum changed the rouge scripts.